### PR TITLE
[OpenACC} Improve diagnostics for 'tag's on clauses/directives

### PR DESCRIPTION
**Pull request by erichkeane**
The 'cache' directive and various clauses have a 'tag' name that is optional.  This patch cleans up the use of the 'cache' version so that we get a nicer diagnostic, and enables us to do the same with clauses in the same situation.

### CODE REVIEW

File: clang/lib/Parse/ParseOpenACC.cpp [hunk 1]
```diff
@@ -164,6 +164,49 @@ bool isOpenACCSpecialToken(OpenACCSpecialTokenKind Kind, Token Tok) {
...
+}
+
+
+/// Parses and consumes an identifer followed immediately by a single colon, and
+/// diagnoses if it is not the 'special token' kind that we require. Used when
+/// the tag is the only valid value.
+/// Return 'true' if the special token was matched, false if no special token,
+/// or an invalid special token was found.
+template <typename DirOrClauseTy>
+bool TryParseAndConsumeSpecialTokenKind(Parser &P, OpenACCSpecialTokenKind Kind,
```
Comment (alexey-bataev) [inline clang/lib/Parse/ParseOpenACC.cpp:188]: tryParseAndConsumeSpecialTokenKind

File: clang/lib/Parse/ParseOpenACC.cpp [hunk 2]
```diff
@@ -164,6 +164,48 @@ bool isOpenACCSpecialToken(OpenACCSpecialTokenKind Kind, Token Tok) {
   llvm_unreachable("Unknown 'Kind' Passed");
 }
 
+// Used for cases where we have a token we want to check against an
+// 'identifier-like' token, but don't want to give awkward error messages in
+// cases where it is accidentially a keyword.
```
Comment (alexey-bataev) [inline clang/lib/Parse/ParseOpenACC.cpp:169]:
```suggestion
/// Used for cases where we have a token we want to check against an
/// 'identifier-like' token, but don't want to give awkward error messages in
/// cases where it is accidentially a keyword.
```

**Review by alexey-bataev — APPROVED** [review]
LG with a nit

**Comment by erichkeane** [comment]
> LG with a nit

Thanks for the super quick reviews!

**Pull request merged**